File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.length()-1;
        int i =0;

        while(n >=0 && s[n] == ' '){
            n--;
        }

        while(n >=0 && s[n] != ' '){
            i++;
            n--;
        }
        return i;
        
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        # s = s.strip().split(" ")
        # print(s)
        # return len(s[-1])

        i = len(s)-1

        while i > 0 and s[i] == " ":
            i -=1
        

        ans = 0

        while i >= 0 and s[i] != " ":
            ans +=1
            i -=1
        
        return ans